### ISSUE TITLE
event loop

### ISSUE THREAD

**Issue by waywardmonkeys** [project-mindy/mindy#65]
We currently have parts of an event loop in the C code within the interpreter. But we aren't using any modern mechanisms ... and it isn't providing support for doing stuff from the Dylan level.

What can we improve here?

Can we learn something from the Haskell [MIO](http://haskell.cs.yale.edu/wp-content/uploads/2013/08/hask035-voellmy.pdf) library?

**Comment by kaveman-** [comment]
libuv? there is also the eventloop from redis (ae.c) that could be used. But first, wouldn't it be better to gather the interpreter globals into a state object?

**Comment by pierredepascale** [comment]
From what I remember all the interpreter state is already encapsulated in a thread_t structure containing the PC, SP, etc...

**Comment by waywardmonkeys** [comment]
Yes, but those then share the same heap and loaded code. There are things where having entirely separate VMs in a single process would be useful (but a fair bit of work).
